test/UnitTest/Microsoft.AspNet.OData.Test.Shared/OpenComplexTypeTest.cs

```diff
@@ -161,6 +167,102 @@ public class OpenComplexTypeTest
         }
 
+        [Fact]
+        public async Task OpenComplexType_PatchNestedComplexTypeProperty()
+        {
+            // Arrange
+            string payload = @"{
+                ""Street"":""UpdatedStreet"",
+                ""Token@odata.type"":""#Guid"",
+                ""Token"":""9B198CA0-9546-4162-A4C0-14EAA255ACA7"",
+                ""BirthDay@odata.type"":""#Date"",
+                ""BirthDay"":""2016-01-29"",
+                ""LineB"": {""Description"": ""DescriptionB""}
+            }";
+            payload = Regex.Replace(payload, @"\s*", "", RegexOptions.Multiline);
+
+            const string requestUri = "http://localhost/odata/OpenCustomers(1)/Address";
+            HttpClient client = GetClient();
+
+            // Act
+            HttpRequestMessage request = new HttpRequestMessage(new HttpMethod("Patch"), requestUri);
+            request.Content = new StringContent(payload);
+            request.Content.Headers.ContentType = MediaTypeWithQualityHeaderValue.Parse("application/json");
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task OpenComplexType_PatchNestedComplexTypeProperty_DoubleNested()
+        {
+            // Arrange
+            string payload = @"{
+                ""Street"":""UpdatedStreet"",
+                ""Token@odata.type"":""#Guid"",
+                ""Token"":""A4D09554-5551-4B36-A1CB-CFBCDB1F4EAD"",
+                ""BirthDay@odata.type"":""#Date"",
+                ""BirthDay"":""2016-01-29"",
+                ""LineA"": {
+                    ""Description"": ""DescriptionA"",
+                    ""PhoneInfo"" : {""ContactName"":""ContactNameA"", ""PhoneNumber"": 7654321}
+                },
+                ""LineB"": {
+                    ""PhoneInfo"": {""ContactName"": ""ContactNameB""}
+                }
+            }";
+            payload = Regex.Replace(payload, @"\s*", "", RegexOptions.Multiline);
+
+            const string requestUri = "http://localhost/odata/OpenCustomers(1)/Address";
+           HttpClient client = GetClient();
+
+            // Act
+            HttpRequestMessage request = new HttpRequestMessage(new HttpMethod("Patch"), requestUri);
+            request.Content = new StringContent(payload);
+            request.Content.Headers.ContentType = MediaTypeWithQualityHeaderValue.Parse("application/json");
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task OpenComplexType_PatchNestedComplexTypeProperty_DeepNestedResourceOnNewSubNode()
```

Comment:
>public async Task OpenComplexType_PatchNestedComplexTypeProperty_DeepNestedResourceOnNewSubNode() [](start = 8, length = 97)

The new added three UT has the similar function body. consider to create a common helper method, and call this method in three UTs?

Reply:
good idea. Create method private async Task ExecutePatchRequest(string payload); used by four tests.

---
In reply to: [190751964](https://github.com/OData/WebApi/pull/1445#discussion_r190751964) [](ancestors = 190751964)